internal/vulnerability/scanner.go

```diff
@@ -201,6 +207,13 @@ func (f VulnerabilityScanner) scan(ctx context.Context, snap ec2.EBSSnapshot) {
 		}
 	}
 
+	go func() {
```

Comment:
I was thinking that we might want to make that `defer` and not go routine for two reasons:

  1. Previously the `f.cleaner.CleanSnapshots(ctx)` was a blocking call to ensure that every "delete snapshot" was done before `VulnerabilityWorker.Run` ended. 
  2. [concern, not an issue currently] Potentially we have <number of snapshots>*routines. Although the snapshots are handled (I believe) serially and we might not end up in an un-capped go routine spawning situation. Still, perhaps it leaves the door open to a situation like this if we modify the snapshot scanning at some point in the future.

Reply:
1. Very good point, moved the goroutine
2. Even though I am not concerned with the resource impact of goroutines (I doubt this will be the bottleneck if we scale to millions of CNVM snapshots), it might be a good idea in case there are rate limits in AWS' side. That said, I feel the goroutine gives a statistically better chance to successfully complete the cleanup as of now.